Source/Core/VideoBackends/D3DCommon/Shader.cpp

```diff
@@ -105,8 +108,9 @@ std::optional<Shader::BinaryData> Shader::CompileShader(D3D_FEATURE_LEVEL featur
   if (FAILED(hr))
   {
     static int num_failures = 0;
-    std::string filename = StringFromFormat(
-        "%sbad_%s_%04i.txt", File::GetUserPath(D_DUMP_IDX).c_str(), target, num_failures++);
+    std::string filename = StringFromFormat("%sbad_%s_%04i_", File::GetUserPath(D_DUMP_IDX).c_str(),
```

Comment:
It might be worth extracting the filename generation logic to a separate function that you can reuse, instead of having to include Common/Version everywhere. Also, it's probably better not to mix StringFromFormat(...) and manual string concatenation. Just add the version and backend name part of the format string

Reply:
Not sure if I did this correctly but I think I properly used StringFromFormat in the latest commit. And the filename format isn't consistent across all cases (see the changes in this PR for OGL). Also, wouldn't we need to include Version.h anyway to avoid indirect includes?

Reply:
>Also, wouldn't we need to include Version.h everywhere anyway to avoid indirect includes?

No, you wouldn't have to expose implementation details in the header.